src/test/ui/generic-associated-types/issue-86218.rs

```diff
@@ -21,7 +18,8 @@ trait Yay<AdditionalValue> {
 
 impl<'a> Yay<&'a ()> for () {
     type InnerStream<'s> = impl Stream<Item = i32> + 's;
-    fn foo<'s>() -> Self::InnerStream<'s> { todo!() }
+    //^ ERROR does not fulfill the required lifetime
+    fn foo<'s>() -> Self::InnerStream<'s> { () }
```

Comment:
I think this indeed should pass, but it fails because the opaque type erroneously captures `'a`. See #96996. So #86218 probably shouldn't be closed.

Reply:
Thinking more about it, now I believe it should pass with this PR even if the opaque type captures `'a` (which is another bug but entirely orthogonal). With this PR we should be able to infer `Opaque<'s, 'a>: 's` even if `'a: 's` doesn't hold, right?

Here's an test using type parameters instead to make it clear. I expect it to pass with this PR but it doesn't:
```rust
#![feature(generic_associated_types)]
#![feature(type_alias_impl_trait)]

pub trait Stream {
    type Item;
}

impl Stream for () {
    type Item = i32;
}

trait Yay<AdditionalValue> {
    type InnerStream<'s>: Stream<Item = i32> + 's;
    fn foo<'s>() -> Self::InnerStream<'s>;
}

impl<I> Yay<I> for () {
    type InnerStream<'s> = impl Stream<Item = i32> + 's;
    fn foo<'s>() -> Self::InnerStream<'s> { () }
}

fn main() {}
```